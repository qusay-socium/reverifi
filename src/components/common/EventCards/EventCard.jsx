import React from 'react';
import { ReactComponent as Add } from 'assets/Icons/add.svg';
import { ReactComponent as Share } from 'assets/Icons/share.svg';
import { ReactComponent as People } from 'assets/Icons/people.svg';
import PropTypes from 'prop-types';
import {
  Card,
  CardHeader,
  CardFooter,
  EventImage,
  EventDate,
  Title,
  EventInfo,
  FooterTop,
  FooterBottom,
  Attendees,
  ListingProperties,
  Property,
} from './event-card.styles';

/**
 * @param   {string}  title     Event title at the bottom of the card component
 * @param   {string}  date      Schedueled date for the event
 * @param   {string}  startTime Time of day scheduled for the event to begin at
 * @param   {string}  endTime   Time of day scheduled for the event to end at
 * @param   {string}  location  Geographical location set for the event
 * @param   {string}  attendees Number of people attending the event
 * @param   {string}  imageUrl  link for the picture associated with the event
 * @param   {string}  eventType Type of event personal or webinar
 * @return {JSX.Element}
 */

export default function EventCard({
  title,
  date,
  startTime,
  endTime,
  location,
  attendees,
  imageUrl,
  eventType,
}) {
  return (
    <Card>
      <CardHeader>
        <EventImage alt="Event" src={imageUrl} />
      </CardHeader>
      <ListingProperties>
        <Property eventType={eventType}>{eventType}</Property>
      </ListingProperties>
      <EventDate>
        <p>{date}</p>
      </EventDate>
      <CardFooter>
        <FooterTop>
          <div>
            <Title>{title}</Title>
            <EventInfo>
              {startTime}-{endTime} @{location}
            </EventInfo>
          </div>
          <div>
            <Add />
            <Share />
          </div>
        </FooterTop>
        <FooterBottom>
          <Attendees>
            <People />
            <p>{attendees}</p>
          </Attendees>
        </FooterBottom>
      </CardFooter>
    </Card>
  );
}
EventCard.propTypes = {
  attendees: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  endTime: PropTypes.string.isRequired,
  eventType: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  startTime: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
