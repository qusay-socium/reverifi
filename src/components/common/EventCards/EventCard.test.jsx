import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import EventCard from './EventCard';
import {
  EventImage,
  Title,
  Property,
  EventInfo,
  FooterTop,
  FooterBottom,
  Attendees,
  EventDate,
} from './event-card.styles';

Enzyme.configure({ adapter: new Adapter() });

let Props;
beforeEach(() => {
  Props = {
    event: [
      {
        attendees: '122',
        date: ' Dec 09 2021',
        endTime: '08:00',
        eventId: 78367,
        eventType: 'N/A',
        id: 1,
        imageUrl:
          'https://images.unsplash.com/photo-1598482327649-e8831e1505be',
        location: 'Area 51',
        startTime: '16:20',
        title: 'Alien Hunt',
      },
    ],
  };
});

describe('Snapshot Testing For Cards', () => {
  it('Should render properly', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const CardTest = shallow(<EventCard />);
    expect(CardTest).toMatchSnapshot();
  });

  it('Should render 1 Image', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const CardTest = shallow(<EventCard {...Props.event} />);
    expect(CardTest.find(EventImage).exists()).toBe(true);
  });
  it('Should render 1 Title', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const CardTest = shallow(<EventCard {...Props.event} />);
    expect(CardTest.find(Title).exists()).toBe(true);
    expect(CardTest.find(Title).type().target).toEqual('p');
  });
  it('Should render 1 Property', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const CardTest = shallow(<EventCard {...Props.event} />);
    expect(CardTest.find(Property).exists()).toBe(true);
    expect(CardTest.find(Property).type().target).toEqual('p');
  });
});
