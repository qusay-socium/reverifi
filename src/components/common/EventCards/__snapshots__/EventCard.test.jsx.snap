// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing For Cards Should render properly 1`] = `ShallowWrapper {}`;
